apps/emqx_ft/src/emqx_ft_storage_exporter_fs.erl

```diff
@@ -265,30 +293,35 @@ start_reader(Options, RelFilepath, CallerPid) ->
 
 %%
 
--spec list(options()) ->
+-spec list(options(), query()) ->
     {ok, [exportinfo(), ...]} | {error, [{node(), _Reason}]}.
-list(_Options) ->
+list(_Options, Query) ->
     Nodes = mria_mnesia:running_nodes(),
-    Replies = emqx_ft_storage_exporter_fs_proto_v1:list_exports(Nodes),
-    {Results, Errors} = lists:foldl(
+    Replies = emqx_ft_storage_exporter_fs_proto_v1:list_exports(Nodes, Query),
+    #{
+        exports := NodeExports,
+        errors := Errors,
+        failures := Failures
+    } = lists:foldl(
         fun
-            ({_Node, {ok, {ok, Files}}}, {Acc, Errors}) ->
-                {Files ++ Acc, Errors};
-            ({Node, {ok, {error, _} = Error}}, {Acc, Errors}) ->
-                {Acc, [{Node, Error} | Errors]};
-            ({Node, Error}, {Acc, Errors}) ->
-                {Acc, [{Node, Error} | Errors]}
+            ({_Node, {ok, {ok, Exports}}}, #{exports := EAcc} = Acc) ->
+                Acc#{exports => [Exports | EAcc]};
+            ({Node, {ok, {error, Reason}}}, #{errors := EAcc} = Acc) ->
+                Acc#{errors => [{Node, Reason} | EAcc]};
+            ({Node, Error}, #{failures := FAcc} = Acc) ->
+                Acc#{failures => [{Node, Error} | FAcc]}
         end,
-        {[], []},
+        #{exports => [], errors => [], failures => []},
         lists:zip(Nodes, Replies)
     ),
-    length(Errors) > 0 andalso
-        ?SLOG(warning, #{msg => "list_remote_exports_failed", errors => Errors}),
-    case Results of
-        [_ | _] ->
-            {ok, Results};
+    length(Failures) > 0 andalso
+        ?SLOG(warning, #{msg => "list_remote_exports_failed", failures => Failures}),
+    Limit = maps:get(limit, Query, -1),
+    case emqx_ft_util:sublist_mergesorted(NodeExports, Limit) of
```

Comment:
It is unobious part, and maybe we need some explanations here. 
`NodeExports` are maps(?), so the order they are sorted does not match the order of folder traversal in the file iterator 🤔 If true, this will cause some transfers to be skipped during the listing 🤔

Reply:
`NodeExports` should be [list](https://github.com/keynslug/emqx/blob/e4356e503da15565a0c5d76b9efa46c598ca02f9/apps/emqx_ft/src/emqx_ft_storage_exporter_fs.erl#L308) of [lists](https://github.com/keynslug/emqx/blob/e4356e503da15565a0c5d76b9efa46c598ca02f9/apps/emqx_ft/src/emqx_ft_storage_exporter_fs.erl#L235) now. Also the [iteration order](https://github.com/keynslug/emqx/blob/e4356e503da15565a0c5d76b9efa46c598ca02f9/apps/emqx_ft/src/emqx_ft_storage_exporter_fs.erl#L209) [should be kept](https://github.com/keynslug/emqx/blob/e4356e503da15565a0c5d76b9efa46c598ca02f9/apps/emqx_ft/src/emqx_ft_fs_iterator.erl#L72). Though this still needs testing.